devices/debian_wifi.py

```diff
@@ -17,3 +18,54 @@ def scan(self):
         self.expect(self.prompt)
 
         return self.before
+
+    def wpa_connect(self,ssid_name,pass_word,wlan_intf):
+
+         '''Generate WPA supplicant file and execute it'''
+
+         self.sendline("rm /etc/"+ssid_name+".conf")
+         self.expect(self.prompt)
+         self.sendline("wpa_passphrase "+ssid_name+ " >> /etc/"+ssid_name+".conf")
+         self.expect("")
+         self.sendline(pass_word)
+         self.expect(self.prompt)
+         self.sendline("cat /etc/"+ssid_name+".conf")
+         self.expect(self.prompt)
+         self.sendline("wpa_supplicant  -B -Dnl80211 -i"+wlan_intf+ " -c/etc/"+ssid_name+".conf")
+         self.expect(self.prompt)
+         wpa_start = None
+         wpa_start = self.before
+         match = re.search('Successfully initialized wpa_supplicant',wpa_start)
```

Comment:
Ok, probably if something has gone wrong in the above command, the wpa_supplicant won't initialise.

Reply:
Yes, if something goes wrong wpa_supplicant wont be intialised . As assertion can be thrown in the testcase based on the return value.